pkgs/dart_mcp/lib/src/api/api.dart

```diff
@@ -225,9 +225,11 @@ extension type RequestId( /*String|int*/ Parameter _) {}
 ///
 /// The receiver must promptly respond, or else may be disconnected.
 ///
-/// The request itself has no parameters and should always be just `null`.
-extension type PingRequest._(Null _) {
+/// The request itself has no parameters.
+extension type PingRequest._(Map<String, Object?> _) implements Request {
   static const methodName = 'ping';
+
+  factory PingRequest() => PingRequest._(const {});
```

Comment:
It looks like this should just implement `Request` now, as it explicitly has the metadata field with the progress token https://github.com/modelcontextprotocol/modelcontextprotocol/blob/main/schema/2025-06-18/schema.json#L1467.

```suggestion
extension type PingRequest._(Map<String, Object?> _) implements Request {
  static const methodName = 'ping';

  factory PingRequest({
    MetaWithProgressToken? meta,
  }) => PingRequest._({
    if (meta != null) '_meta': meta,
  });
```